library/drumhat.py

```diff
@@ -151,9 +165,26 @@ def led_off(pad):
     led = LEDMAP[idx]
     dh.set_led_state(led, False)
 
-for x in range(8):
-    dh.on(x,event='press',   handler=_handle_press)
-    dh.on(x,event='release', handler=_handle_release)
+def ensure_setup():
```

Comment:
Needs to drop `ensure_setup()` and just have `setup()` set an `_is_setup` flag and not run any of the actual setup routines twice. The current method is unnecessarily convoluted.